charmcraft/main.py

```diff
@@ -123,6 +123,13 @@ class ProvideHelpException(Exception):
         "--quiet",
         "Only show warnings and errors, not progress",
     ),
+    GlobalArgument(
+        "trace",
+        "flag",
+        "-t",
+        "--trace",
+        "Show all information needed to trace any internal behaviour",
```

Comment:
Maybe "any" isn't strictly true, only those that have been instrumented will display information.

Reply:
Improved, thanks